importlib_metadata/__init__.py

```diff
@@ -199,16 +199,31 @@ def __iter__(self):
         warnings.warn(msg, DeprecationWarning)
         return iter((self.name, self))
 
-    def __reduce__(self):
-        return (
-            self.__class__,
-            (self.name, self.value, self.group),
-        )
-
     def matches(self, **params):
         attrs = (getattr(self, param) for param in params)
         return all(map(operator.eq, params.values(), attrs))
 
+    def _key(self):
+        return tuple(getattr(self, key) for key in 'name value group'.split())
```

Comment:
```suggestion
        return self.name, self.value, self.group
```
Shorter, faster and more comprehensible